Added Mach-O module ref Issue #30

zcatbear opened the pull request:
Mach-O parsing has been added and tested against several mach-o files.

jxb5151 commented:
This is great! Thanks for this contribution. 

A couple of comments upon first glance, could you:
- Ensure '#!/usr/bin/env python' is at the start of the module and disposition files.
- Also commit an updated rules.yara file with your signature included.
- Consider adding the Apache license preamble to your module with your name.
- Remove .pyc file from the commit.
- Does this make sense as a META_ module rather than an EXTRACT_? Appears we are enumerating lots of good macho metadata here.

zcatbear commented:
- Added `#!/usr/bin/env python` to start of module and disposition
- Updated yara rules
- Apache license added to module
- Removed .pyc
- Changed name to META module and reflected that change in disposition

compsecmonkey commented:
Thanks again for the contribution @zcatbear !

I have been testing the new module and have a few additional observations as I continue to test.
- There is a reference to the old "EXTRACT_MACHO" in  /modules/**init**.py which prevents FSF from running the module.
- Consider removing the macholibre provided MD5 / SHA1 / SHA256 data points as those are are provided by the META_BASIC_INFO module. 
- Consider removing the macholibre provide overall size for the same reason as the hashes.
